consul/consul
# Improves admin banners section

Where
=====
* **Related Issue:** #1626 (this PR closes #1626)

What
====
Improve the generation of a banner adding two color-pickers to the form so that the admin can choose the background color and the font color separately.
Some checkboxes were added also to let the admin choose the web sections where each banner will appear.

How
===
- Add two HTML5 color picker form input, one for the background color and another one for the font color. Some text_fields were added also to each picker, because it wont work correctly in IE.
- Add the checkboxes to select the web sections where the banner will appear. For this, a new DB table has been added to store the sections.
- Now, banners show the choosen color for the background and for the font.

Screenshots
===========
![banner01](https://user-images.githubusercontent.com/31625251/34941978-35392efc-f9f6-11e7-9506-2291e85d771b.png)

Test
====
- Feature specs to test the color changing when selecting from the color-picker or when writting in the text-field.
- The color selection has been added to the generation of the banner. 

Deployment
==========
- Run rake task `rake web_sections:generate` to fill the web_sections table.
- Run rake task `rake banners:migrate_style` to migrate the banner `style` column to `background_color` and `font_color`, so that the banners won't appear all white.

Warnings
========
- `dev_seeds` where updated to include the generation of the web_sections.

Deprecation warnings
=
The Changelog of v0.13 should include two warnings:
- Columns `style` and `image` from `banners` table will be removed in v0.14. Run `rake banners:migrate_style` to migrate the colors to `background_color` and `font_color`.
- Banner images (app/assets/images/banners) will be removed in v0.14, as they won't be used anymore. If you want to keep them, backup them manually.
